zebra/zapi_msg.c

```diff
@@ -481,17 +481,14 @@ void nbr_connected_add_ipv6(struct interface *ifp, struct in6_addr *address)
 	IPV6_ADDR_COPY(&p.u.prefix6, address);
 	p.prefixlen = IPV6_MAX_PREFIXLEN;
 
-	ifc = listnode_head(ifp->nbr_connected);
-	if (!ifc) {
-		/* new addition */
-		ifc = nbr_connected_new();
-		ifc->address = prefix_new();
-		ifc->ifp = ifp;
-		listnode_add(ifp->nbr_connected, ifc);
-	}
-
+	/* new addition */
```

Comment:
the relationship between rtadv.c and zapi_msg.c seems a little off: `process_advert` checks for an existing address, instead of doing that check here. then zapi_msg.c has a "delete_ipv6" that can remove something from this nbr_connected list, but that isn't ever called?

Reply:
yes. it seems the rtadv server is far from being production ready yet.
I don't pretend to fix all.
I wanted to check that topotests was passing ( a priori no regression with other daemons)
this said, zsend_interface_addresses() may be called on several points, and having more than one nbr may impact the daemons
so I would not push it to master only with that fix.